packages/sit-onyx/src/components/OnyxSwitch/OnyxSwitch.vue

```diff
@@ -0,0 +1,274 @@
+        background-color: var(--onyx-color-base-primary-200);
+      }
+
+      .onyx-switch__position {
+        background-color: var(--onyx-color-base-background-blank);
+
+        .onyx-switch__icon {
+          color: var(--onyx-color-text-icons-primary-soft);
+        }
+      }
+    }
+
+    &:disabled + .onyx-switch__container {
+      .onyx-switch__track {
+        background-color: var(--onyx-color-base-neutral-200);
+      }
+
+      .onyx-switch__position {
+        background-color: var(--onyx-color-base-neutral-300);
+
+        .onyx-switch__icon {
+          color: var(--onyx-color-text-icons-neutral-inverted);
+        }
+      }
+    }
+
+    &--invalid + .onyx-switch__container {
+      .onyx-switch__track {
+        background-color: var(--onyx-color-base-danger-200);
+        border: var(--onyx-1px-in-rem) solid var(--onyx-color-base-danger-500);
+        height: 1.375rem;
+        width: 2.25rem;
+      }
+
+      .onyx-switch__position {
+        background-color: var(--onyx-color-base-danger-500);
+
+        .onyx-switch__icon {
+          color: var(--onyx-color-text-icons-neutral-inverted);
+        }
+      }
+    }
+
+    &--invalid:checked + .onyx-switch__container {
+      .onyx-switch__track {
+        background-color: var(--onyx-color-base-danger-500);
+      }
+
+      .onyx-switch__position {
+        background-color: var(--onyx-color-base-background-blank);
+
+        .onyx-switch__icon {
+          color: var(--onyx-color-text-icons-danger-intense);
+        }
+      }
+    }
+
+    &--readonly + .onyx-switch__container {
+      .onyx-switch__track {
+        background-color: var(--onyx-color-base-neutral-200);
+      }
+
+      .onyx-switch__position {
+        background-color: var(--onyx-color-base-neutral-300);
+
+        .onyx-switch__icon {
+          color: var(--onyx-color-text-icons-neutral-inverted);
+        }
+      }
+    }
+
+    &--readonly:checked + .onyx-switch__container {
+      .onyx-switch__track {
+        background-color: var(--onyx-color-base-primary-200);
+      }
+
+      .onyx-switch__position {
+        background-color: var(--onyx-color-base-background-blank);
+
+        .onyx-switch__icon {
+          color: var(--onyx-color-text-icons-primary-soft);
+        }
+      }
+    }
+  }
+
+  &__container {
+    position: relative;
+    display: block;
+    margin: var(--onyx-spacing-2xs);
+    width: 2.375rem;
+    height: 1.5rem;
+
+    .onyx-switch__track {
+      position: absolute;
+      width: 100%;
+      height: 1.5rem;
+      background-color: var(--onyx-color-base-neutral-300);
+      border-radius: var(--onyx-radius-full);
+      transition: background-color 0.2s ease;
+    }
+
+    .onyx-switch__position {
+      position: absolute;
+      display: flex;
+      flex-wrap: wrap;
+      align-content: normal;
+      left: 0.125rem;
+      top: 0.125rem;
+      right: 0.125rem;
+      bottom: 0.125rem;
+      width: 1.25rem;
+      height: 1.25rem;
```

Comment:
Please make use of flex `justify-content` and `align-items`, as well as `padding`. Absolute positioning must be avoided.